cmd/root.go

```diff
@@ -187,29 +186,49 @@ func AddCommands(cmd *cobra.Command) {
-		config.TortoiseBeacon.FirstVotingRoundDurationMs, "First voting round duration in milliseconds")
-	cmd.PersistentFlags().IntVar(&config.TortoiseBeacon.VotingRoundDurationMs, "tortoise-beacon-voting-round-duration-ms",
-		config.TortoiseBeacon.VotingRoundDurationMs, "Voting round duration in milliseconds")
-	cmd.PersistentFlags().IntVar(&config.TortoiseBeacon.WeakCoinRoundDurationMs, "tortoise-beacon-weak-coin-round-duration-ms",
-		config.TortoiseBeacon.WeakCoinRoundDurationMs, "Weak coin round duration in milliseconds")
-	cmd.PersistentFlags().IntVar(&config.TortoiseBeacon.WaitAfterEpochStart, "tortoise-beacon-wait-after-epoch-start-ms",
+	cmd.PersistentFlags().StringVar(&q, "tortoise-beacon-q",
+		config.TortoiseBeacon.Q.String(), "Ratio of dishonest spacetime (for calculating ATX threshold). It should be a string representing a rational number.")
+	cmd.PersistentFlags().Uint64Var(&roundsNumber, "tortoise-beacon-rounds-number",
+		uint64(config.TortoiseBeacon.RoundsNumber), "Amount of rounds in every epoch")
+	cmd.PersistentFlags().DurationVar(&config.TortoiseBeacon.GracePeriodDuration, "tortoise-beacon-grace-period-duration",
+		config.TortoiseBeacon.GracePeriodDuration, "Grace period duration in milliseconds")
+	cmd.PersistentFlags().DurationVar(&config.TortoiseBeacon.ProposalDuration, "tortoise-beacon-proposal-duration",
+		config.TortoiseBeacon.ProposalDuration, "Proposal duration in milliseconds")
+	cmd.PersistentFlags().DurationVar(&config.TortoiseBeacon.FirstVotingRoundDuration, "tortoise-beacon-first-voting-round-duration",
+		config.TortoiseBeacon.FirstVotingRoundDuration, "First voting round duration in milliseconds")
+	cmd.PersistentFlags().DurationVar(&config.TortoiseBeacon.VotingRoundDuration, "tortoise-beacon-voting-round-duration",
+		config.TortoiseBeacon.VotingRoundDuration, "Voting round duration in milliseconds")
+	cmd.PersistentFlags().DurationVar(&config.TortoiseBeacon.WeakCoinRoundDuration, "tortoise-beacon-weak-coin-round-duration",
+		config.TortoiseBeacon.WeakCoinRoundDuration, "Weak coin round duration in milliseconds")
+	cmd.PersistentFlags().DurationVar(&config.TortoiseBeacon.WaitAfterEpochStart, "tortoise-beacon-wait-after-epoch-start",
 		config.TortoiseBeacon.WaitAfterEpochStart, "How many milliseconds to wait after a new epoch is started.")
-	cmd.PersistentFlags().Float64Var(&config.TortoiseBeacon.Theta, "tortoise-beacon-theta",
-		config.TortoiseBeacon.Theta, "Ratio of votes for reaching consensus")
-	cmd.PersistentFlags().IntVar(&config.TortoiseBeacon.VotesLimit, "tortoise-beacon-votes-limit",
+	cmd.PersistentFlags().StringVar(&theta, "tortoise-beacon-theta",
+		config.TortoiseBeacon.Theta.String(), "Ratio of votes for reaching consensus")
+	cmd.PersistentFlags().Uint64Var(&config.TortoiseBeacon.VotesLimit, "tortoise-beacon-votes-limit",
 		config.TortoiseBeacon.VotesLimit, "Maximum allowed number of votes to be sent")
 
+	var ok bool
+
+	config.TortoiseBeacon.Q, ok = new(big.Rat).SetString(q)
```

Comment:
i think using RatVar for Q and Theta that implements pflag.Value will be cleaner

Reply:
In this case, a wrapper-type is needed, which looks less clean to me. But I'll think of it

Reply:
i think wrapper can be applied only when Q is registered e.g:

```go
cmd.VarP((*RatVar)(config.TortoiseBeacon.Q), ...)
```

Reply:
Good idea, thanks.